internal/provider/search_attribute_data_source.go

```diff
@@ -0,0 +1,147 @@
+package provider
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	"github.com/hashicorp/terraform-plugin-framework/datasource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+	"go.temporal.io/api/enums/v1"
+	"go.temporal.io/api/operatorservice/v1"
+	"google.golang.org/grpc"
+)
+
+// Ensures that SearchAttributeDataSource fully satisfies the datasource.DataSource and
+// datasource.DataSourceWithConfigure interfaces.
+var (
+	_ datasource.DataSource              = &SearchAttributeDataSource{}
+	_ datasource.DataSourceWithConfigure = &SearchAttributeDataSource{}
+)
+
+// NewSearchAttributeDataSource returns a new instance of the SearchAttributeDataSource.
+func NewSearchAttributeDataSource() datasource.DataSource {
+	return &SearchAttributeDataSource{}
+}
+
+// SearchAttributeDataSource implements the Terraform data source interface for Temporal SearchAttributes.
+type SearchAttributeDataSource struct {
+	client operatorservice.OperatorServiceClient
+}
+
+// SearchAttributeDataSourceModel defines the structure for the data source's configuration and read data.
+type SearchAttributeDataSourceModel struct {
+	Name      types.String `tfsdk:"name"`
+	Type      types.String `tfsdk:"type"`
+	Namespace types.String `tfsdk:"namespace"`
+}
+
+// Metadata sets the metadata for the Temporal SearchAttribute data source, specifically the type name.
+func (d *SearchAttributeDataSource) Metadata(ctx context.Context, req datasource.MetadataRequest, resp *datasource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_search_attribute"
+}
+
+// Schema defines the schema for the Temporal SearchAttribute data source.
+func (d *SearchAttributeDataSource) Schema(ctx context.Context, req datasource.SchemaRequest, resp *datasource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		// This description is used by the documentation generator and the language server.
+		MarkdownDescription: "Temporal Search Attribute resource",
+
+		Attributes: map[string]schema.Attribute{
+			"name": schema.StringAttribute{
+				MarkdownDescription: "Search Attribute Name",
+				Required:            true,
+			},
+			"type": schema.StringAttribute{
+				MarkdownDescription: "Search Attribute Indexed Value Type, which defines the type of data stored in the attribute",
+				Computed:            true,
+			},
+			"namespace": schema.StringAttribute{
+				MarkdownDescription: "Namespace with which the Search Attribute is associated",
+				Required:            true,
```

Comment:
+1

Reply:
Unlike the resource schema, data source schema does not support default strings. Because of this, I kept the namespace required since all the api calls require a namespace. We can add logic to make it optional and then manually set it to 'default' for every api call if the namespace is not set, but that seems more messy than requiring a namespace from the user. @dipali-bhatt @parinmaru2 Thoughts?

Reply:
If it is marked as required, then all users will have to provide it and that doesn't match with the API's behavior. If for API it is optional, then we should also make it optional. If that requires us setting the namespace as default in our logic, then we should do that.

Reply:
@dipali-bhatt It is not optional for the api - all the api calls require a namespace. That's why we will need to manually hadcode "default" everywhere if we make the namespace optional

Reply:
@amibaid Looking at this [API documentation](https://pkg.go.dev/go.temporal.io/api@v1.30.1/operatorservice/v1#AddSearchAttributesRequest), doesn't look like namespace is a required attribute. I believe that is what we discussed in the past as well. Has anything changed after we discussed this the last time?

Reply:
Whenever I have tried calling the API without a namespace or with `namespace = ""` I get the following error: 
```Search attribute creation failed: rpc error: code = InvalidArgument desc = Failed to add search attributes to store sqlite: Namespace is not set on request.```
So it seems the namespace is required

Reply:
I have raised a support thread here to investigate the discrepancy, but for now treating namespace as required by API and supplying `default` when user doesn't provide a namespace

Reply:
Update: It seems like namespace is required, so the current functionality should be ok
[Slack support thread](https://temporalio.slack.com/archives/CTDTU3J4T/p1721258119875649?thread_ts=1721255485.197359&cid=CTDTU3J4T)